test/test_api/test_api_announcement.py

```diff
@@ -0,0 +1,316 @@
+# -*- coding: utf8 -*-
+# This file is part of PYBOSSA.
+#
+# Copyright (C) 2017 Scifabric LTD.
+#
+# PYBOSSA is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# PYBOSSA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with PYBOSSA.  If not, see <http://www.gnu.org/licenses/>.
+import json
+import io
+from default import db, with_context
+from test_api import TestAPI
+from factories import AnnouncementFactory
+from factories import UserFactory
+from pybossa.repositories import AnnouncementRepository
+from mock import patch
+from nose.tools import assert_raises
+
+announcement_repo = AnnouncementRepository(db)
+
+
+class TestAnnouncementAPI(TestAPI):
+
+    @with_context
+    def test_query_announcement(self):
+        """Test API query for announcement endpoint works"""
+        owner = UserFactory.create()
+        user = UserFactory.create()
+        announcements = AnnouncementFactory.create_batch(9)
+        announcement = AnnouncementFactory.create()
+
+        # As anon
+        url = '/api/announcement'
+        res = self.app_get_json(url)
+        data = json.loads(res.data)
+        assert len(data) == 10, data
+
+        # As user
+        res = self.app_get_json(url + '?api_key=' + user.api_key)
+        data = json.loads(res.data)
+        # TODO: project_id is tried to load with api_key. Do not load it?
+        assert data['exception_cls'] == 'AttributeError', data
```

Comment:
This should return all the announcements, correct?

Reply:
Basically, you need to remove the owner_id from the context manager to ensure this works.